indent/ruby.vim

```diff
@@ -322,10 +336,16 @@ function! s:DeindentingKeyword(cline_info)
       let msl  = s:GetMSL(line('.'))
       let line = getline(line('.'))
 
-      if strpart(line, 0, col('.') - 1) =~ '=\s*$' &&
```

Comment:
Take the substring from the beginning to the character before the `DeindentingKeyword` (in this specific case) and see if it ends in `=\s*` (is the leading substring an assignment).